lib/public/systemtag/isystemtagsmanager.php

```
@@ -0,0 +1,109 @@
+	 *
+	 * @param array|string $tagIds The ID or array of IDs of the tags to retrieve
+	 *
+	 * @return \OCP\SystemTag\ISystemTag[] array of system tags or empty array if none found
+	 *
+	 * @since 9.0.0
+	 */
+	public function getTagsById($tagIds);
+
+	/**
+	 * Returns the tag object matching the given attributes.
+	 *
+	 * @param string $tagName tag name
+	 * @param bool $userVisible whether the tag is visible by users
+	 * @param bool $userAssignable whether the tag is assignable by users
+	 *
+	 * @return \OCP\SystemTag\ISystemTag system tag
+	 *
+	 * @throws \OCP\Files\NotFoundException if tag does not exist
+	 *
+	 * @since 9.0.0
+	 */
+	public function getTag($tagName, $userVisible, $userAssignable);
+
+	/**
+	 * Creates the tag object using the given attributes.
+	 *
+	 * @param string $tagName tag name
+	 * @param bool $userVisible whether the tag is visible by users
+	 * @param bool $userAssignable whether the tag is assignable by users
+	 *
+	 * @return \OCP\SystemTag\ISystemTag system tag
+	 *
+	 * @throws \OCP\Files\AlreadyExistsException if tag already exists
+	 *
+	 * @since 9.0.0
+	 */
+	public function createTag($tagName, $userVisible, $userAssignable);
+
+	/**
+	 * Returns all known tags, optionally filtered by visibility.
+	 *
+	 * @param bool $visibleOnly whether to only return user visible tags
+	 * @param string $nameSearchPattern optional search pattern for the tag name
+	 *
+	 * @return \OCP\SystemTag\ISystemTag[] array of system tags or empty array if none found
+	 *
+	 * @since 9.0.0
+	 */
+	public function getAllTags($visibleOnly = false, $nameSearchPattern = null);
+
+	/**
+	 * Updates the given tag
+	 *
+	 * @param string $tagId tag id
+	 * @param string $newName the new tag name
+	 * @param bool $userVisible whether the tag is visible by users
+	 * @param bool $userAssignable whether the tag is assignable by users
+	 *
+	 * @throws \OCP\Files\AlreadyExistsException if tag already exists
```

Should also throw if tag_id was not found